app/scripts/modules/core/src/managed/StatusBubbleStack.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { take } from 'lodash';
+
+import { IStatusBubbleProps, StatusBubble } from './StatusBubble';
+
+import './StatusBubbleStack.less';
+
+export interface IStatusBubbleStackProps {
+  borderColor: string;
+  maxStatusBubbleCount: number;
+  statuses: Array<Pick<IStatusBubbleProps, 'iconName' | 'appearance'>>;
+}
+
+export const StatusBubbleStack = ({ borderColor, maxStatusBubbleCount, statuses }: IStatusBubbleStackProps) => {
+  // We only have limited space in the UI and we cannot show all the statuses in the stack, so only take the first few
+  // important statuses.
+  const statusBubblesToRender = take(
+    statuses,
+    statuses.length <= maxStatusBubbleCount ? maxStatusBubbleCount : maxStatusBubbleCount - 1,
+  );
+  const hiddenStatusBubbleCount = statuses.length - statusBubblesToRender.length;
+
+  return (
+    <div className="flex-container-h middle sp-margin-s-right">
+      {statusBubblesToRender.map((status, index) => (
+        <div
+          className="StatusBubbleContainer"
+          key={status.iconName}
+          style={{
+            backgroundColor: borderColor,
+            zIndex: statusBubblesToRender.length - index,
+          }}
+        >
+          <StatusBubble iconName={status.iconName} appearance={status.appearance} size="small" quantity="2" />
+        </div>
+      ))}
+      {hiddenStatusBubbleCount && (
+        <div className="HiddenStatusCount" key="count">
```

Comment:
is the `key` necessary here now that it's not being rendered inside a `map()` call?